pkg/controller/remotekubernetescluster/controller.go

```diff
@@ -263,3 +273,39 @@ func (rkcc *Controller) enqueueRemoteKubernetesClusterUsingSecret(secret *corev1
 		return rkc.Spec.KubeconfigSecretRef.Namespace == secret.Namespace && rkc.Spec.KubeconfigSecretRef.Name == secret.Name
 	}))
 }
+
+func (rkcc *Controller) addScyllaDBCluster(obj interface{}) {
+	rkcc.handlers.HandleAdd(
+		obj.(*scyllav1alpha1.ScyllaDBCluster),
+		rkcc.enqueueScyllaDBCluster,
+	)
+}
+
+func (rkcc *Controller) updateScyllaDBCluster(old, cur interface{}) {
+	rkcc.handlers.HandleUpdate(
+		old.(*scyllav1alpha1.ScyllaDBCluster),
+		cur.(*scyllav1alpha1.ScyllaDBCluster),
+		rkcc.enqueueScyllaDBCluster,
+		rkcc.deleteScyllaDBCluster,
+	)
+}
+
+func (rkcc *Controller) deleteScyllaDBCluster(obj interface{}) {
+	rkcc.handlers.HandleDelete(
+		obj,
+		rkcc.enqueueScyllaDBCluster,
+	)
+}
+
+func (rkcc *Controller) enqueueScyllaDBCluster(depth int, obj kubeinterfaces.ObjectInterface, op controllerhelpers.HandlerOperationType) {
```

Comment:
the name of the func is misleading, in the end you are enqueuing an RKC - should it say `enqueueRemoteKubernetesClustersReferencedByScyllaDBCluster`  or something along these lines?